### PR TITLE
Migrate `pdb-buildinfo-cl-cmd` and `pgo-indirect-call-promotion` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Please try:

try-job: x86_64-msvc
try-job: x86_64-mingw
try-job: i686-msvc
try-job: i686-mingw
try-job: x86_64-gnu-llvm-17
try-job: aarch64-apple

### CODE REVIEW

**Comment by Oneirical** [comment]
Invalid UTF-8 error. Maybe this will fare better, but I have doubts.

`assert_contains(String::from_utf8_lossy(&rfs::read("my_crate_name.pdb")), env_var("RUSTC"));`

@rustbot ready

Please run try again, if you think this stands a shot.

**Comment by jieyouxu** [comment]
> Invalid UTF-8 error. Maybe this will fare better, but I have doubts.
> 
> `assert_contains(String::from_utf8_lossy(&rfs::read("my_crate_name.pdb")), env_var("RUSTC"));`

Shouldn't we use something like `assert_invalid_utf8_contains`, because AFAIK pdb debug info files don't need to be UTF-8 (and it wouldn't make sense if it did).

**Comment by ChrisDenton** [comment]
Yeah. PDB is a binary format so it's highly unlikely to be valid UTF-8 and doing a lossy conversion seems like a lot of unnecessary work. Basically this should read the file as bytes (`fs::read`) then search within it using `env_var("RUSTC").as_bytes()` or some helper that does that.

**Comment by jieyouxu** [comment]
Yeah we could use something like [`bstr::ByteSlice::find_iter`](https://docs.rs/bstr/latest/bstr/trait.ByteSlice.html#method.find_iter).

@rustbot author

**Comment by Oneirical** [comment]
Okay, I gave the `bstr` implementation a try, but I am of little faith - I tried to run this locally without the `.pdb` extension on a Linux system and failed at the RUSTC assertion. It's possible that the `pdb` file on MSVC is different and that this is the whole point of the test, though.

@rustbot review

File: tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    assert!(&rfs::read("my_crate_name.pdb").find(env_var("RUSTC").as_bytes()).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""main.rs""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""-g""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""--crate-name""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""my_crate_name""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""--crate-type""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""bin""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""-C""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""metadata=dc9ef878b0a48666""#).is_some());
+    assert!(&rfs::read("my_crate_name.pdb").find(br#""--out-dir""#).is_some());
```
Comment (jieyouxu) [inline tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs:30]: Suggestion: a helper like `assert_pdb_contains`, probably just in this test file.

**Comment by jieyouxu** [comment]
> Okay, I gave the `bstr` implementation a try, but I am of little faith - I tried to run this locally without the `.pdb` extension on a Linux system and failed at the RUSTC assertion. It's possible that the `pdb` file on MSVC is different and that this is the whole point of the test, though.

Isn't pdb files a microsoft concept?

**Comment by Oneirical** [comment]
I fixed the problem that caused the MSVC failure ("-C metadata" instead of "-Cmetatadata") and I have added the local helper function. Please run try again.

@rustbot review

**Comment by jieyouxu** [comment]
Thanks!
@bors r+ rollup=iffy (pdb, pgo)

**Comment by tgross35** [comment]
@bors r-

Merge conflict, I don't know why bors didn't r- yet (it has been a few hours)

**Comment by Oneirical** [comment]
Rebase complete, please re-approve. @rustbot ready

**Comment by tgross35** [comment]
@Oneirical github still says there is a conflict in `allowed_run_make_makefiles` (maybe another one merged?)

**Comment by Oneirical** [comment]
> @Oneirical github still says there is a conflict in `allowed_run_make_makefiles` (maybe another one merged?)

Yes, it was caused by #128757 which just got merged.

**Comment by Oneirical** [comment]
This is very weird, my Github branch [here](https://github.com/Oneirical/rust/blob/not-to-be-undertestimated/src/tools/tidy/src/allowed_run_make_makefiles.txt) is different from the [files changed](https://github.com/rust-lang/rust/blob/bf89c8c14bc97a4ab8bb8c9a567c241675bada35/src/tools/tidy/src/allowed_run_make_makefiles.txt) tab in this PR.

Note the `run-make/pgo-gen-lto/Makefile`.

**Comment by Oneirical** [comment]
???

**Comment by Oneirical** [comment]
I'm not too sure what happened there, but it seems like everything is back the way it should, and that the ghost merge conflict is gone.

Please re-approve.

**Comment by tgross35** [comment]
Maybe you rebased onto the local `master` without pulling first? In any case, looks like it worked now.

@bors r=jieyouxu

**Comment by tgross35** [comment]
This merged already, why did it get picked up?

@bors retry

**Comment by tgross35** [comment]
Bors didn't correctly mark a couple PRs as merged, isn't `r-`ing merge conflicts, and doesn't seem to be picking anything new up. I'm going to do a resync to see if it sorts things.

@bors treeclosed=100

**Pull request merged**